tests/e2e/actions.go

```diff
@@ -2862,26 +2862,162 @@ func (tr Chain) GetPathNameForGorelayer(chainA, chainB ChainID) string {
 	return pathName
 }
 
-// Run an instance of the Hermes relayer using the "evidence" command,
-// which detects evidences committed to the blocks of a consumer chain.
+// detect evidences committed to the blocks of a consumer chain
+// either by running an instance of the Hermes relayer using the "evidence" command,
+// or by queyring manually the consumer chain.
 // Each infraction detected is reported to the provider chain using
 // either a SubmitConsumerDoubleVoting or a SubmitConsumerMisbehaviour message.
 type StartConsumerEvidenceDetectorAction struct {
-	Chain ChainID
+	Chain     ChainID
+	Submitter ValidatorID
 }
 
 func (tr Chain) startConsumerEvidenceDetector(
 	action StartConsumerEvidenceDetectorAction,
+	useRelayer bool,
 	verbose bool,
 ) {
 	chainConfig := tr.testConfig.chainConfigs[action.Chain]
-	// run in detached mode so it will keep running in the background
-	bz, err := tr.target.ExecDetachedCommand(
-		"hermes", "evidence", "--chain", string(chainConfig.ChainId)).CombinedOutput()
-	if err != nil {
-		log.Fatal(err, "\n", string(bz))
+	// the Hermes doesn't support evidence handling for Permissionless ICS yet
+	// TODO: @Simon refactor once https://github.com/informalsystems/hermes/pull/4182 is merged.
+	if useRelayer {
+		// run in detached mode so it will keep running in the background
```

Comment:
If I understood correctly, it seems that if `!useRelayer`, we do not run this in the background but check if there's currently evidence. If so, it might make sense to rename `startConsumerEvidenceDetector` to something like `detectEvidence...`. If that's the case, maybe we should run Hermes in a similar way here and  not in detached mode.

Reply:
Good point! Renamed to` detectConsumerEvidenceAction`